src/olympia/users/tests/test_admin.py

```diff
@@ -44,13 +46,34 @@ def test_search_for_multiple_users(self):
         self.client.login(email=user.email)
         response = self.client.get(
             self.list_url,
-            {'q': '%s,%s' % (self.user.pk, another_user.pk)},
+            {'q': '%s,%s,foobaa' % (self.user.pk, another_user.pk)},
             follow=True)
         assert response.status_code == 200
         doc = pq(response.content)
         assert str(self.user.pk) in doc('#result_list').text()
         assert str(another_user.pk) in doc('#result_list').text()
 
+    def test_search_for_multiple_user_ids(self):
+        """Test the optimization when just searching for matching ids."""
+        user = user_factory()
+        another_user = user_factory()
+        self.grant_permission(user, 'Admin:Tools')
+        self.grant_permission(user, 'Users:Edit')
+        self.client.login(email=user.email)
+        with CaptureQueriesContext(connection) as queries:
+            response = self.client.get(
+                self.list_url,
+                {'q': '%s,%s' % (self.user.pk, another_user.pk)},
+                follow=True)
+            queries = '; '.join(q['sql'] for q in queries.captured_queries)
+            in_sql = f'`users`.`id` IN ({self.user.pk}, {another_user.pk})'
+            print(queries)
+            assert in_sql in queries
```

Comment:
might be worth asserting on the amount of queries too when you already capture it.

Reply:
https://github.com/mozilla/addons-server/pull/11388/commits/951e8a185d17c48e5c165f873d308d234e6c5b66